ic-cdk/src/call.rs

```diff
@@ -163,7 +281,35 @@ pub enum CallError {
     CandidDecodeFailed(String),
 }
 
+/// The error type for inter-canister calls.
+#[derive(Debug, Clone)]
+pub struct SystemError {
+    /// See [`RejectCode`].
+    pub reject_code: RejectCode,
+    /// The reject message.
+    ///
+    /// When the call was rejected asynchronously (IC rejects the call after it was enqueued),
+    /// this message is set with [`msg_reject`](crate::api::msg_reject).
+    ///
+    /// When the call was rejected synchronously (`ic0.call_preform` returns non-zero code),
+    /// this message is set to a fixed string ("failed to enqueue the call").
+    pub reject_message: String,
+    /// See [`ErrorCode`].
+    ///
+    /// # Note
+    ///
+    /// As of the current version of the IC, the error codes are not available in the system API.
+    /// Please DO NOT rely on the error codes until they are officially supported.
+    pub error_code: ErrorCode,
```

Comment:
What's the purpose of having this 'pub' when the value is generally incorrect (from reject_to_error) and the error codes haven't been officially finalized?

Reply:
We can remove the pub for now. But when ic0.msg_error_code is officially supported, we will need to add back the pub. Then that will be a breaking change.

One alternative is to make the fields private and provide public getters. But that will harm the UX when users want to extract multiple fields.

After more thinking, I feel that the alternative approach above could be better. I will followup with Martin.